streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -76,7 +76,7 @@
     private static final String TOSTREAM_NAME = "KTABLE-TOSTREAM-";
 
 
-    public final ProcessorSupplier<K, ?> processorSupplier;
+    public final ProcessorSupplier<?, ?> processorSupplier;
```

Comment:
This is because in KStreamImpl, `reduceByKey` and `aggregateByKey` with windows specs returns a `KTable<Windowed<K>, ?>` and others returns a `KTable<K, ?>`, hence the key type cannot be bound to K any more.